product_dimension_volumetric_weight/models/product.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    weight_volumetric = fields.Float(
+        string="Volumetric weight",
+        compute="_compute_weight_volumetric",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends("dimensional_uom_id", "volume")
+    def _compute_weight_volumetric(self):
+        for item in self:
+            item.weight_volumetric = (
+                item.volume * item.dimensional_uom_id.weight_volumetric_ratio
+            )
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    weight_volumetric = fields.Float(
+        string="Volumetric weight",
+        compute="_compute_weight_volumetric",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends("dimensional_uom_id", "volume")
+    def _compute_weight_volumetric(self):
```

Comment:
No need to repeat this method if it's the same as upstream `product.template` one.